File: app/src/main/java/com/codebosses/comical/adapters/SearchComicAdapter.java
package com.codebosses.comical.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.codebosses.comical.R;
import com.codebosses.comical.databinding.RowSearchComicBinding;
import com.codebosses.comical.endpoints.EndpointKeys;
import com.codebosses.comical.pojo.Comic;
import com.codebosses.comical.pojo.event_bus.EventBusAdapterClick;

import org.greenrobot.eventbus.EventBus;

import java.util.ArrayList;
import java.util.List;

public class SearchComicAdapter extends RecyclerView.Adapter<SearchComicAdapter.SearchComicHolder> {

    private Context context;
    private LayoutInflater layoutInflater;
    private List<Comic> comicsList = new ArrayList<>();

    public SearchComicAdapter(Context context, List<Comic> comicsList) {
        this.context = context;
        this.comicsList = comicsList;
        layoutInflater = LayoutInflater.from(context);
    }

    @NonNull
    @Override
    public SearchComicHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new SearchComicHolder(DataBindingUtil.inflate(layoutInflater, R.layout.row_search_comic, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull SearchComicHolder holder, int position) {
        holder.searchComicBinding.setComic(comicsList.get(position));
    }

    @Override
    public int getItemCount() {
        return comicsList.size();
    }

    class SearchComicHolder extends RecyclerView.ViewHolder {

        RowSearchComicBinding searchComicBinding;

        public SearchComicHolder(@NonNull RowSearchComicBinding itemView) {
            super(itemView.getRoot());
            this.searchComicBinding = itemView;

            itemView.getRoot().setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    EventBus.getDefault().post(new EventBusAdapterClick(getAdapterPosition(), EndpointKeys.SEARCH_COMIC_CLICK));
                }
            });
        }
    }
}

File: app/src/main/java/com/codebosses/comical/ui/main/search/SearchComicAdapter.kt
package com.codebosses.comical.ui.main.search

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.codebosses.comical.R
import com.codebosses.comical.repository.eventbus.EventBusSearchClick
import com.codebosses.comical.repository.model.comics.ComicResult
import com.codebosses.comical.repository.model.search.SearchResult
import com.codebosses.comical.utils.extensions.gone
import com.codebosses.comical.utils.extensions.visible
import kotlinx.android.synthetic.main.row_search_comic.view.*
import org.greenrobot.eventbus.EventBus

class SearchComicAdapter(val context: Context, val searchList: List<SearchResult>) : RecyclerView.Adapter<SearchComicAdapter.SearchComicHolder>() {

    private val layoutInflater = LayoutInflater.from(context)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SearchComicHolder {
        return SearchComicHolder(layoutInflater.inflate(R.layout.row_search_comic, parent, false))
    }

    override fun getItemCount(): Int {
        return searchList.count()
    }

    override fun onBindViewHolder(holder: SearchComicHolder, position: Int) {
        holder.bindData(searchList[position])
    }

    inner class SearchComicHolder(val view: View) : RecyclerView.ViewHolder(view) {
        fun bindData(searchResult: SearchResult) {
            Glide.with(context)
                    .load(searchResult.comic_banner_path)
                    .centerCrop()
                    .placeholder(R.drawable.banner_placeholder)
                    .into(view.imageViewComicSearchRow)
            view.textViewNameSearchComicRow.text = searchResult.comic_name
            view.textViewDescriptionSearchComic.text = searchResult.summary
            if (searchResult.is_favorite == 0) {
                view.imageViewFavoriteSearchComic.gone()
            } else {
                view.imageViewFavoriteSearchComic.visible()
            }

            view.setOnClickListener {
                EventBus.getDefault().post(EventBusSearchClick(adapterPosition))
            }
        }
    }

}